lib/form_data.js

```diff
@@ -406,8 +405,7 @@ FormData.prototype.submit = function(params, cb) {
 
   // use custom params
   } else {
-
-    options = populate(params, defaults);
+    options = populate(JSON.parse(JSON.stringify(params)), defaults);
```

Comment:
Could `Object.assign` be used here? Cloning an object by de/re-serialising seems inefficient.

Reply:
Object.assign doesn't deep-clone, and my assumption is in some cases the input params can have referenced objects? I may be wrong on this.

Reply:
It seems crude to use JSON for this purpose. Perhaps make note in the docs that this object is expected to be mutated?

Reply:
In the end, after looking at the code, I actually think there is no expectation of a referenced value within the expected `params` object, so `Object.assign` might be sufficient. I'll double-check and make the change.

Reply:
There's no other way to deep-clone as efficiently as this in pure JavaScript. In general, this is not happening in some hot loop - it's in the submit function and will happen asynchronously relatively rarely. And most callers are probably using a string type `params` rather than an object for this, avoiding this path altogether.

I hesitate to just let the behavior stay - generally, libraries shouldn't be mutating unless callers explicitly expect it, e.g. in other languages by passing in a mutable reference as opposed to immutable. That's actually why we considered this a bug upstream and filed a PR, with a temporary fix on our side of just passing in a fresh object each time we submit the form.

Reply:
^ GitHub got my comments upside down with their timestamps somehow. Edited to fix order.

Also checked and Object.assign should be good.